tests/queries/0_stateless/02366_union_header_conversion.sql

```diff
@@ -0,0 +1 @@
+SELECT * FROM (SELECT sqrt(0.5) AS a UNION ALL SELECT sqrt(dummy IN (SELECT 0))) ORDER BY a;
```

Comment:
BTW it is not only about `UNION`, but about sub queries:

```sql
SELECT * FROM (SELECT identity(dummy IN (SELECT 0)));
Code: 47. DB::Exception: Missing columns: 'identity(in(dummy, _subquery1))' while processing query: 'SELECT `identity(in(dummy, _subquery1))` FROM (SELECT identity(dummy IN (SELECT 0)))', required columns: 'identity(in(dummy, _subquery1))' 'identity(in(dummy, _subquery1))'. (UNKNOWN_IDENTIFIER)
```

So better to reflect this in the test case name.